core/src/main/java/nu/yona/server/subscriptions/service/BuddyService.java

```diff
@@ -201,20 +207,53 @@ public BuddyDto addBuddyToAcceptingUser(UserDto acceptingUser, BuddyConnectReque
 
 		acceptingUser.assertMobileNumberConfirmed();
 		Buddy buddy = buddyRepository.save(createBuddyEntity(connectRequestMessageEntity));
-		BuddyDto buddyDto = createBuddyDto(buddy);
+		BuddyDto buddyDto = createBuddyDtoForNewBuddy(buddy);
 		userService.addBuddy(acceptingUser, buddyDto);
 		return buddyDto;
 	}
 
 	private BuddyDto createBuddyDto(Buddy buddy)
 	{
 		Optional<UserAnonymizedDto> buddyUserAnonymizedDto = getUserAnonymizedDto(buddy);
-		return BuddyDto.createInstance(buddy, buddyUserAnonymizedDto);
+		BuddyAnonymizedDto buddyAnonymizedDto = getBuddyAnonymized(buddy);
+		return BuddyDto.createInstance(buddy, buddyAnonymizedDto, buddyUserAnonymizedDto);
+	}
+
+	private BuddyDto createBuddyDtoForNewBuddy(Buddy buddy)
+	{
+		Optional<UserAnonymizedDto> buddyUserAnonymizedDto = buddy.getUserAnonymizedId()
+				.map(userAnonymizedService::getUserAnonymized);
+		BuddyAnonymizedDto buddyAnonymizedDto = BuddyAnonymizedDto.createInstance(buddy.getBuddyAnonymized());
+		return BuddyDto.createInstance(buddy, buddyAnonymizedDto, buddyUserAnonymizedDto);
+	}
+
+	private BuddyAnonymizedDto getBuddyAnonymized(Buddy buddy)
+	{
+		return getBuddyAnonymized(buddy, getOwningUserAnonymized(buddy));
+	}
+
+	private BuddyAnonymizedDto getBuddyAnonymized(Buddy buddy, UserAnonymizedDto owningUserAnonymizedDto)
+	{
+		return getOwningUserAnonymized(buddy).getBuddyAnonymized(buddy.getBuddyAnonymizedId());
```

Comment:
`owningUserAnonymizedDto` is not used here. Also I think we can inline this method into the method above because it is only used once.

Reply:
The parameter was already removed in a later commit. This method is also used from ``getUserOfBuddy`` (line 765).